forgejo-runner: 11.3.1 -> 12.2.0

stv0g opened the pull request:
Bump forgejo-runner from v11.3.1 to v12.1.1

## Things done

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin (broken)
  - [ ] aarch64-darwin (broken)
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [x] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

stv0g committed:
forgejo-runner: 11.3.1 -> 12.2.0

Changelog:
- https://code.forgejo.org/forgejo/runner/releases/tag/v12.0.0
- https://code.forgejo.org/forgejo/runner/releases/tag/v12.0.1
- https://code.forgejo.org/forgejo/runner/releases/tag/v12.1.0
- https://code.forgejo.org/forgejo/runner/releases/tag/v12.1.1
- https://code.forgejo.org/forgejo/runner/releases/tag/v12.1.2
- https://code.forgejo.org/forgejo/runner/releases/tag/v12.2.0

tebriel commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 468244`
Commit: `376f363743aa66a0bbd73f981e31071119b203c8`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>forgejo-runner</li>
  </ul>
</details>

tebriel commented:
I don't think forgejo-runner supports aarch64 based on https://code.forgejo.org/forgejo/runner#architectures-os and that there's no aarch64 binaries on the release page.

tebriel reviewed (changes requested):
Please add a note to the release notes that [12.0.0](https://code.forgejo.org/forgejo/runner/releases/tag/v12.0.0) has a breaking change that requires `git` to be installed on the system as it's no longer inbuilt to the binary.

stv0g commented:
I've rebased and pushed again

emilylange commented on pkgs/by-name/fo/forgejo-runner/package.nix:
```diff
@@ -42,17 +43,17 @@ let
 buildGoModule rec {
   pname = "forgejo-runner";
-  version = "11.3.1";
+  version = "12.1.1";
```
https://code.forgejo.org/forgejo/runner/releases/tag/v12.1.2 has been released since.

adamcstephens committed:
forgejo-runner: add backward compatibility wrapper

git is now an external requirement

emilylange commented on pkgs/by-name/fo/forgejo-runner/package.nix:
```diff
@@ -64,21 +68,27 @@ buildGoModule rec {
     # also calls it `forgejo-runner`
     mv $out/bin/runner $out/bin/forgejo-runner
+
+    # provide backward compatbility since v12 removed bundled git
```
Nit: There has never been a bundled git. It's always been `go-git`. What's new is that in addition to `go-git`, `git` is now required for certain codepaths.

emilylange commented on pkgs/by-name/fo/forgejo-runner/package.nix:
```diff
@@ -64,21 +68,27 @@ buildGoModule rec {
     mv $out/bin/runner $out/bin/forgejo-runner
+
+    # provide backward compatbility since v12 removed bundled git
+    wrapProgram $out/bin/forgejo-runner --prefix PATH : ${lib.makeBinPath [ git ]}
```
Not strictly an issue, but I would like to point out, that `--prefix` shadows whatever other `git` may already be in the `$PATH`.

For example, `nixos/gitea-actions-runner` adds `gitMinimal` to the systemd unit `$PATH` by default when using `:host`, but our wrapped `git` will take precedence over that `gitMinimal`.

I feel like `--suffix` might be better here, especially in the spirit of backwards compatibility. But I can also see how this discussion could go both ways.

```suggestion
    wrapProgram $out/bin/forgejo-runner --suffix PATH : ${lib.makeBinPath [ git ]}
```

Not blocking.

emilylange reviewed (approved):
I would have probably squashed the two version bumps and I don't see much value in the comment diffs (*docker install* vs *Docker daemon* and *Fix* vs *fix*), but overall this PR is fine as-is.

Left 3 suggestions, but all of them are non-blocking.

stv0g commented:
I have addressed @emilylange's comments:

- Squashed the two version bump commits
- Replaced `--prefix` with `--suffix` in the wrapper
- Replaced `git` with `gitMinimal`

Changes are force pushed.

nrabulinski commented:
Can we get this rebased now that Darwin is unbroken and build there too?
